query/src/sessions/context_shared.rs

```diff
@@ -91,6 +94,11 @@ impl DatabendQueryContextShared {
             source_abort_handle.abort();
         }
 
+        let http_query = self.http_query.read();
+        if let Some(handle) = &*http_query {
+            handle.abort();
```

Comment:
need this because to abort the channel send/recv between the task generate datastream and handle.
and may later add query_id or something to find HTTP_query associate from the session/query  if needed

@zhang2014

Reply:
There may be no need new abort stream for http handler(All sources are also abortable).

Reply:
currently to make sure kill success, I`ll try to think of a better solution later.